### PR TITLE
[patch] Fix use cache bug

### PR DESCRIPTION
**Pull request by kyfujisa**
I modified the timing of the return when using the cache.

If it is cached, it will be authorized before any certificate related validation can be run.

### CODE REVIEW

File: authorizerd.go [hunk 1]
```diff
@@ -364,6 +360,10 @@ func (a *authorizer) verify(ctx context.Context, m mode, tok, act, res string, c
 
 	switch m {
 	case roleToken:
+		if isCacheHit {
```
Comment (kyfujisa) [inline authorizerd.go:363]: I' m doing this because even if I functionalize it, still need to write an IF statement.

File: authorizerd.go [hunk 2]
```diff
@@ -351,11 +351,7 @@ func (a *authorizer) verify(ctx context.Context, m mode, tok, act, res string, c
 	}
 
 	// check if exists in verification success cache
-	_, ok := a.cache.Get(tok + act + res)
-	if ok {
-		glg.Debugf("use cached result. tok: %s, act: %s, res: %s", tok, act, res)
-		return nil
-	}
+	_, isCacheHit := a.cache.Get(tok + act + res)
```
Comment (WindzCUHK) [inline authorizerd.go:354]:
```suggestion
	_, isCacheHit := a.cache.Get(tok + string(cert.Raw) + act + res)
```

suggestion: update key structure, keep current logic

Comment (kyfujisa) [reply authorizerd.go:354]: ok, I' ll consider using thumbprint.

Comment (kyfujisa) [reply authorizerd.go]: f301fa1

File: authorizerd.go [hunk 3]
```diff
@@ -350,8 +352,21 @@ func (a *authorizer) verify(ctx context.Context, m mode, tok, act, res string, c
 		return errors.Wrap(ErrInvalidParameters, "empty action / resource")
 	}
 
+	var certThumbprint string
 	// check if exists in verification success cache
-	_, isCacheHit := a.cache.Get(tok + act + res)
+	if cert != nil {
+		ct := sha256.Sum256(cert.Raw)
+		certThumbprint = hex.EncodeToString(ct[:])
+		if _, ok := a.cache.Get(tok + act + res + certThumbprint); ok {
```
Comment (WindzCUHK) [inline authorizerd.go:360]: 1. seems better to refactor as a `getCacheKey` func.
2. latency (use thumbprint) V.S. memory (use raw)

**Comment by ssunorz** [comment]
新しくPRを作りなおしたので、このPRはクロースします
https://github.com/yahoojapan/athenz-authorizer/pull/75

**Pull request closed without merging**